config/crd/patches/cainjection_in_assign.yaml

```diff
@@ -0,0 +1,8 @@
+# The following patch adds a directive for certmanager to inject CA into the CRD
+# CRD conversion requires k8s 1.13 or later.
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
```

Comment:
Did we remove permissions entirely?

I thought we were going to add them to the gatekeeper-manager role?

Reply:
After digging deeper I learned that there is a convention where [controller-gen](https://github.com/open-policy-agent/gatekeeper/blob/0bf7afd45254c1f0052d7e401fcf296c73429af9/Makefile#L149) generates these permissions and outputs them to [role.yaml](https://github.com/open-policy-agent/gatekeeper/blob/master/config/rbac/role.yaml) according to kubebuilder markers on the controllers themselves (and not on CRDs) as we can see for [config](https://github.com/open-policy-agent/gatekeeper/blob/0bf7afd45254c1f0052d7e401fcf296c73429af9/pkg/controller/config/config_controller.go#L176) and [constraints](https://github.com/open-policy-agent/gatekeeper/blob/0bf7afd45254c1f0052d7e401fcf296c73429af9/pkg/controller/constraint/constraint_controller.go#L185) for example.

I can add them manually to gatekeeper-manager role (which will be overridden and exclude these permission when we run deploy) or we can add them to the role.yaml file in the controller related PRs (I already added a comment for one [here](https://github.com/open-policy-agent/gatekeeper/pull/927#discussion_r517575035))

Reply:
@maxsmythe  was the permissions comment/question placed here on purpose ? (cainjection_in_assign.yaml) I'm trying to understand, out of placement context, if I missed anything else.

Reply:
It was just placed randomly, I probably should have just added it to the bottom of the PR. Sorry.

You are correct. I'd forgotten that we were putting the permissions on the controller (which makes sense as that's the thing that actually needs the access).